src/PatientBrowserTutorial/index.html

```diff
@@ -0,0 +1,208 @@
+                console.log("token type is:" + response.tokenType);
+            }
+        }
+    }
+    
+    function signIn() {
+        myMSALObj.loginPopup(requestObj).then(function (loginResponse) {
+            showWelcomeMessage();
+            document.getElementById('fileSelector')[0].style.visibility = "visible";
+            document.getElementById('myTable').style.visibility = "visible";
+            myMSALObj.acquireTokenSilent(requestObj).then(function (tokenResponse) {
+                callFHIRServer(FHIRConfig.FHIRendpoint + '/Patient','GET',null, tokenResponse.accessToken, FHIRCallback);
+            }).catch(function (error) {
+                console.log(error);
+                if (requiresInteraction(error.errorCode)) {
+                    myMSALObj.acquireTokenPopup(requestObj).then(function (tokenResponse) {
+                        callFHIRServer(FHIRConfig.FHIRendpoint + '/Patient','GET',null, tokenResponse.accessToken, FHIRCallback);
+                    }).catch(function(error) {
+                        console.log(error);
+                    })
+                }
+            });
+        }).catch(function(error) {
+            console.log(error);
+        })
+    }
+
+    function requiresInteraction(errorCode) {
+        if (!errorCode || !errorCode.length) {
+            return false;
+        }
+        return errorCode === "consent_required" ||
+            errorCode === "interaction_required" ||
+            errorCode === "login_required";
+    }
+
+    function signOut() {
+        myMSALObj.logout();
+    }
+
+    function FHIRCallback(json) {
+        for (var i = 0; i < Object.keys(json.entry).length; i++) {
+            json2table(json, i);
+            var display = JSON.stringify(json.entry[i].resource, null, 2);
+            document.getElementById('View_Patient_'+i).setAttribute('onclick', 'showPatientEntries('+display+','+i+')');
+        }
+    }
+
+    function callFHIRServer(theUrl, method, message, accessToken, callBack) {
+        var xmlHttp = new XMLHttpRequest();
+        xmlHttp.onreadystatechange = function () {
+            if (this.readyState == 4 && this.status == 200)
+                callBack(JSON.parse(this.responseText));
+        }
+        xmlHttp.open(method, theUrl, true);
+        xmlHttp.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
+        xmlHttp.setRequestHeader('Authorization', 'Bearer ' + accessToken);
+        xmlHttp.send(message);
+    }
+
+    function FHIRProcessBundle(data) {
+        myMSALObj.acquireTokenSilent(requestObj).then(function(tokenResponse) {
+            for (var i=0; i<Object.keys(data.entry).length; i++) {
+                if (data.entry[i].resource.resourceType == "Patient") {
+                    resource = JSON.stringify(data.entry[i].resource);
+                    callFHIRServer(FHIRConfig.FHIRendpoint + '/' + data.entry[i].resource.resourceType, 'POST', resource, tokenResponse.accessToken, loadPatients());
```

Comment:
This should be a `PUT` to the resource with the correct `id`. Otherwise if you keep uploading the same file, it will add duplicate entries.